### PR TITLE
CR-1135432 fix decimal positions for electrical report

### PR DESCRIPTION
**Pull request by rajkumar-xilinx**
Signed-off-by: Rajkumar Rampelli <rampelli@amd.com>

#### Problem solved by the commit
Using static_cast<double> shows the value in decimal points, but it is not 3 digit precision by default. So, fixed decimal point issue using std::setprecision() in sensor code for electrical report.
#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
CR-1135432
#### How problem was solved, alternative solutions (if any) and why they were rejected
Added changes in common sensor code to fix this issue.
#### Risks (if any) associated the changes in the commit
NA
#### What has been tested and how, request additional testing if necessary
examine's electrical report
#### Documentation impact (if any)
NA

### CODE REVIEW

**Comment by rajkumar-xilinx** [comment]
**Without this PR**
before

**With this PR**
image

File: src/runtime_src/core/common/sensor.cpp [hunk 1]
```diff
@@ -167,9 +168,12 @@ read_data_driven_electrical(const std::vector<xq::sdm_sensor_info::data_type>& c
      * So, actual sensor value => 12000 * 10 ^ (-3) = 12 Volts.
      */
     auto unitm = pow(10, tmp.unitm);
-    pt.put("voltage.volts", static_cast<double>(tmp.input) * unitm);
-    pt.put("voltage.max", static_cast<double>(tmp.max) * unitm);
-    pt.put("voltage.average", static_cast<double>(tmp.average) * unitm);
+    stream << std::fixed << std::setprecision(3) << static_cast<double>(tmp.input) * unitm;
+    pt.put("voltage.volts", stream.str());
+    stream << std::fixed << std::setprecision(3) << static_cast<double>(tmp.max) * unitm;
```
Comment (chvamshi-xilinx) [inline src/runtime_src/core/common/sensor.cpp:173]: Don't you need to reset the stream? It gets concatenated to existing stream object.

Comment (rajkumar-xilinx) [reply src/runtime_src/core/common/sensor.cpp:174]: taken care of this. Thank you

Comment (chvamshi-xilinx) [reply src/runtime_src/core/common/sensor.cpp:174]: Better to write simple function to get the value. I see the same line in 6 different places.

**Comment by dayeh-xilinx** [comment]
retest this please.

File: src/runtime_src/core/common/sensor.cpp (hunk 1 above)
Comment (rajkumar-xilinx) [reply src/runtime_src/core/common/sensor.cpp:174]: In format_base10_shiftdown3() API, decimal_shift value is 1000 and it is constant. In our case, this value may not be 1000 and this value is derived from unitm for us.
So, I can't directly use this function here.

**Review by chvamshi-xilinx — APPROVED** [review]

**Comment by rajkumar-xilinx** [comment]
retest this please

File: src/runtime_src/core/common/sensor.cpp (hunk 1 above)
Comment (chvamshi-xilinx) [reply src/runtime_src/core/common/sensor.cpp]: As discussed, i was asking you to create one more generic function. Latest changes looks good

File: src/runtime_src/core/common/utils.cpp [hunk 2]
```diff
@@ -21,6 +21,7 @@
 #include "utils.h"
 #include <atomic>
 #include <cstdint>
+#include <cmath>
```
Comment (keryell) [inline src/runtime_src/core/common/utils.cpp:24]: Sort.

**Review by rozumx — APPROVED** [review]
Glad to see that the data is correctly be fixed in the underlying property tree.

While this isn't an issue, in the future the reports might be needing a different precision value.  If this should occur, please update the report's format routines and NOT how the values are added to property tree.

File: src/runtime_src/core/common/utils.cpp (hunk 2 above)
Comment (rajkumar-xilinx) [reply src/runtime_src/core/common/utils.cpp:23]: Done

**Comment by dayeh-xilinx** [comment]
retest this please.

**Pull request merged**